Monika After Story/game/event-handler.rpy

```diff
@@ -1628,6 +1623,78 @@ init python:
         # otherwise we didnt unlock anything because nothing available
         return False
 
+#TODO:Need to block talk, extra and play from being pressed while eventlabel inactivity_detection is being called/jumped to.
+init 10 python:
+    import time
+    global mas_afk_time
+    global mas_afk_duration
+    global mas_afk_already
+    mas_afk_already = False
+    # How much time in seconds should pass, before it's considered that player is afk.
+    mas_afk_duration = 600
+    mas_afk_time = time.time() + mas_afk_duration
+    config.overlay_screens.append("inactivity_detection_screen")
+    class MASInactivity(renpy.Displayable):
+        
+        import pygame
+
+        def __init__(self):
+            super(MASInactivity, self).__init__()
+
+        def event(self, ev, x, y, st):
+            global mas_afk_time
+            global mas_afk_duration
+            # Constant checking for player activity - mouse movement, keystrokes and mouse button clicks
+            if ev.type == pygame.MOUSEMOTION or ev.type == pygame.KEYDOWN or ev.type == pygame.MOUSEBUTTONDOWN:
+                mas_afk_time = time.time() + mas_afk_duration
+            if mas_afk_time < time.time():
+                mas_afk_time = time.time() + 10000000000
+                # Checking if event can be restarted.
+                if not mas_isRstBlk(persistent.current_monikatopic):
+                    pushEvent(persistent.current_monikatopic)
+                    pushEvent('continue_event')
+                    persistent.current_monikatopic = 0
+                    #if can be restarted jump to afk dialogue.
+                    renpy.jump("inactivity_detection")
+                else:
+                #if can't be restarted simply call afk dialogue.
+                    renpy.call("inactivity_detection")
+            renpy.redraw(self, 0)
+
+        def render(self, width, height, st, at):
+            return renpy.Render(0, 0)
+
+screen inactivity_detection_screen():
+    add MASInactivity()
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="inactivity_detection",
+            category=["dev"],
+            prompt="INACTIVITY DETECTION (ACTUALLY NO NEED TO PRESS THAT)",
+            pool=False,
+            unlocked=False
+        )
+    )
+
+label inactivity_detection:
+    #TODO: I'm terrible at making dialogues, can someone please check if this is actually alright?
+    $ global mas_afk_already
+    if not mas_afk_already:
+        $ mas_afk_already = True
+        m "Are you still there?"
+        m "You are back, [player]!"
+        $ mas_afk_already = False
+    else:
+        #Only repeating first line, because call to inactivity for the second time detection skipped it. 
+        #This is going to be a lot harder if there will be more lines than 2. But I already know how to fix it.
+        m "Are you still there?"
+        $ mas_afk_already = False
```

Comment:
we have a custom screen called `mas_background_timed_jump`, which auto jumps to a label after a set time.

In this case, you should do a menu using the timed jump. [example of use](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/script-story-events.rpy#L490)

you should have it jump to another label that just returns "quit", which will then close the game safely.

I can add this later if you can't figure it out. If so, mark a TODO here for timed screen so I can do that later.